apps/comps/hooks/useAuth.ts

```diff
@@ -121,10 +143,25 @@ export const useUserSession = (): UserSessionState => {
     }
   }, [profileIsSuccess, profileData, setAuthState]);
 
-  return {
-    user: authState.user,
+  const sessionState = useMemo<UserSessionState>(() => {
+    if (!isClient) {
+      return { isInitialized: false };
+    }
+
+    return {
+      isInitialized: true,
+      user: authState.user,
+      isAuthenticated,
+      isProfileUpdated,
+      isLoading: profileIsLoading,
+    };
+  }, [
+    isClient,
+    authState.user,
     isAuthenticated,
     isProfileUpdated,
-    isLoading: profileIsLoading,
-  };
+    profileIsLoading,
+  ]);
+
+  return sessionState;
```

Comment:
This solves the problem for all calls to useUserSession, although I'm not completely convinced by it. I also considered persisting the auth/user state in a cookie, that would allow SSR to access it, although it could introduce other issues. Plus, we’d need to modify which user fields are persisted, since it wouldn’t be feasible to send all of them in a cookie